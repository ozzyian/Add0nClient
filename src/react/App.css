.center {
  position: absolute;
  left: 20%;
  margin-left: -50px;
  top: 50%;
  margin-top: -50px;
  width: 70%;
}

.center-spinner {
  position: absolute;
  left: 50%;
  margin-left: -50px;
  top: 50%;
  margin-top: -50px;
}

#container {
  margin-top: 2em;
}
