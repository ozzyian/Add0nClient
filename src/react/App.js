import React from 'react';
import AddonTable from './components/AddonTable';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';
const ipc = require('electron').ipcRenderer;

/**
 *
 */
class App extends React.Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.state = {initiated: false, addons: []};
  }

  /**
   *
   */
  async componentDidMount() {
    const res = await ipc.invoke('app-load');
    if (!res) {
      return;
    } else {
      this.setState({initiated: true, addons: res});
    }
  }
  /**
   *
   * @param {Object} event
   */
  async onChange(event) {
    const path = event.target.files[0].path.split('\\');
    path.pop();
    const addons = await ipc.invoke('init-with-path', path.join('/'));
    this.setState({initiated: true, addons: addons});
  }
  /**
   *
   * @param {*} e
   */

  /**
   * @return {*}
   */
  render() {
    if (this.state.initiated) {
      return (
        <AddonTable
          initiated={this.state.initiated}
          addons={this.state.addons}
        ></AddonTable>
      );
    } else {
      return (
        <div className="custom-file center">
          <input
            type="file"
            className="custom-file-input"
            id="customFile"
            onChange={(e) => this.onChange(e)}
          ></input>
          <label
            className="custom-file-label"
            htmlFor="customFile"
            overflow="false"
          >
            Choose Wow.exe or ClassicWoW.exe
          </label>
        </div>
      );
    }
  }
}

export default App;
