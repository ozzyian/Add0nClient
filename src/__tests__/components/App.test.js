import React from 'react';
import {shallow} from 'enzyme';
import App from '../../react/App';

describe('Snapshot tests', () => {
  it('it renders correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when file path is choosen', () => {
    const component = shallow(<App />);
    component.setState({initiated: true, addons: []});
    expect(component).toMatchSnapshot();
  });
});
