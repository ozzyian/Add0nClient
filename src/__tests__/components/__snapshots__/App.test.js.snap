// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests it renders correctly 1`] = `
<div
  className="custom-file center"
>
  <input
    className="custom-file-input"
    id="customFile"
    onChange={[Function]}
    type="file"
  />
  <label
    className="custom-file-label"
    htmlFor="customFile"
    overflow="false"
  >
    Choose Wow.exe or ClassicWoW.exe
  </label>
</div>
`;

exports[`Snapshot tests renders correctly when file path is choosen 1`] = `
<AddonTable
  addons={Array []}
  initiated={true}
/>
`;
